src/Kmd.Logic.Email.Client/EmailClient.cs

```diff
@@ -199,13 +199,38 @@ public async Task<IEnumerable<EmailProviderConfiguration>> GetAllProviderConfigu
                 {
                     case System.Net.HttpStatusCode.OK:
                         return configurationDetailsResponse.Body;
-
                     default:
                         throw new EmailException(configurationDetailsResponse?.Body?.ToString() ?? "Invalid configuration provided to access Email service");
                 }
             }
         }
 
+        /// <summary>
+        /// Get the information about email request.
+        /// </summary>
+        /// <param name="emailRequestDetails">Email request details.</param>
+        /// <returns>EmailDetailsResponse.</returns>
+        public async Task<EmailDetailsResponse> GetEmailDetails(EmailRequestDetails emailRequestDetails)
+        {
+            var client = this.CreateClient();
+
+            using (var emailDetailsResponse = await client.GetEmailDetailsWithHttpMessagesAsync(
+                 emailRequestDetails.SubscriptionId, emailRequestDetails.RequestId).ConfigureAwait(false))
+            {
+                switch (emailDetailsResponse?.Response?.StatusCode)
+                {
+                    case System.Net.HttpStatusCode.OK:
+                        return emailDetailsResponse.Body;
+
+                    case System.Net.HttpStatusCode.NotFound:
+                        return null;
+
+                    default:
+                        throw new EmailException(emailDetailsResponse?.Body?.ToString() ?? "Error accessing Email details service.");
```

Comment:
Message can be `Error accessing email service`.

Reply:
Changed .